DisguiseUnityRenderStream/Runtime/DisguiseCameraCapture.cs

```diff
@@ -23,67 +29,107 @@ public void Start()
             m_cameraData = new CameraData();
 
             m_camera = GetComponent<Camera>();
-            StreamDescription stream = Array.Find(m_RenderStream.Streams, s => s.name == gameObject.name);
-            m_frameSender = new FrameSender(gameObject.name, m_camera, stream);
-            RenderPipelineManager.endFrameRendering += RenderPipelineManager_endFrameRendering;
+            var stream = Array.Find(m_RenderStream.Streams, s => s.name == gameObject.name);
+            m_frameSender = new FrameSender(gameObject.name, stream);
+
+            m_capture = gameObject.AddComponent<CameraCapture>();
+            m_capture.hideFlags |= HideFlags.HideAndDontSave;
+
+            m_capture.description = m_frameSender.description;
+            m_capture.onCapture += OnCapture;
+        }
+
+        void OnEnable()
+        {
+            m_capture.enabled = true;
         }
 
-        // Update is called once per frame
-        public void Update()
+        void OnDisable()
         {
-            // set tracking
-            if (m_RenderStream == null) return;
+            m_capture.enabled = false;
+        }
+
+        // TODO: inject somewhere early in the player loop?
+        void Update()
+        {
+            if (m_RenderStream == null)
+                return;
+            
+            m_newFrameData = m_RenderStream.HasNewFrameData &&
+                             m_frameSender != null &&
+                             m_frameSender.GetCameraData(ref m_cameraData);
             
-            m_newFrameData = m_RenderStream.HasNewFrameData && m_frameSender != null && m_frameSender.GetCameraData(ref m_cameraData);
-            float cameraAspect = m_camera.aspect;
-            Vector2 lensShift = new Vector2(0.0f, 0.0f);
+            UpdateCamera(
+                m_camera,
+                m_newFrameData ? m_cameraData : null,
+                m_frameSender?.subRegion
+            );
+        }
+        
+        void OnCapture(ScriptableRenderContext context, CameraCapture.Capture capture)
+        {
             if (m_newFrameData)
             {
-                cameraAspect = m_cameraData.sensorX / m_cameraData.sensorY;
-                if (m_cameraData.cameraHandle != 0)  // If no camera, only set aspect
+                m_frameSender?.SendFrame(m_RenderStream.LatestFrameData, m_cameraData, capture.cameraTexture);
+                m_newFrameData = false;
+            }
+        }
+
+        static void UpdateCamera(Camera cam, in CameraData? cameraData, in Rect? cameraSubRegion)
```

Comment:
I don't think the `in` params here saves you any copying. Since, `Nullable<T>` is not a readonly type, it will still make defensive copies.

Reply:
In my head it was for readability, like `const` to signify that those parameters don't change in the long function ahead. Is there anything similar in C# that would be correct?

Reply:
Unfortunately, there's no equivalent of the C++ `const` in C# that I'm aware of.